go/flagutil/flagutil_test.go

```diff
@@ -82,27 +78,69 @@ func TestStringMap(t *testing.T) {
 		},
 	}
 	for _, want := range wanted {
-		if err := v.Set(want.in); err != want.err {
-			t.Errorf("v.Set(%v): %v", want.in, want.err)
-			continue
-		}
+		assert.Equal(t, want.err, v.Set(want.in))
 		if want.err != nil {
 			continue
 		}
 
 		if len(want.out) != len(v) {
-			t.Errorf("want %#v, got %#v", want.out, v)
+			assert.Equal(t, want.out, v)
```

Comment:
Delete the above `if len(want.out) != len(v) {` condition. Make `assert.Equal` stand on its own.

Reply:
`assert.Equal(t, want.out, v)` should go outside the `if len(want.out) != len(v)` statement.